app/services/preserved_object_handler_results.rb

```diff
@@ -107,25 +103,29 @@ def report_results
         candidate_workflow_results << r
       end
     end
-    stack_trace = caller(1..1).first[/`\S+/].chop[1..-1]
+    stack_trace = caller(0..1).join("\n")
     report_errors_to_workflows(candidate_workflow_results, stack_trace)
     result_array
   end
 
+  private
+
+  def result_hash(code, msg_args=nil)
+    { code => result_code_msg(code, msg_args) }
+  end
+
   def report_errors_to_workflows(candidate_workflow_results, stack_trace)
     return if candidate_workflow_results.empty?
     value_array = []
-    candidate_workflow_results.each do |x|
-      x.each_value do |val|
+    candidate_workflow_results.each do |results|
+      results.each_value do |val|
```

Comment:
is this "results" or a single "result" ?

Reply:
its an array of hashes, where it could be 1 result (if it a moab-validation error) or multiple ( if it is anything else in the preservation-audit step)

Reply:
isn't `candidate_workflow_results` the array of hashes?   and isn't the variable currently called `results` actually a single `result_hash`?

Names matter.

Reply:
oh you're right, i had to print everything out. i changed it to `result_hash` makes it easier to understand

Reply:
good catch